packages/react-composites/src/composites/common/utils.tsx

```diff
@@ -60,3 +60,18 @@ export const useMinMaxDragPosition = (modalLayerHostId: string, rtl?: boolean):
  * @private
  */
 export const defaultSpokenLanguage = 'en-us';
+
+/* @conditional-compile-remove(breakout-rooms) */
+/**
+ * @private
+ */
+export const busyWait = async (checkCondition: () => boolean, retryLimit?: number): Promise<void> => {
+  const delayMs = 500;
+  let retryCount = 0;
+  while (!checkCondition()) {
+    await new Promise((resolve) => setTimeout(resolve, delayMs));
+    if (retryLimit && retryCount++ >= retryLimit) {
```

Comment:
nit: if retryLimit is zero this will fail

Reply:
It should break when it is `retryLimit` is 0 because `retryCount` would be >= than 0 on the first iteration. If you mean when retryLimit is undefined which would be the default then yes it would run forever. Should I prevent that and set a default retryLimit?

Reply:
> It should break when it is retryLimit is 0 because retryCount would be >= than 0

This won't happen, when it's 0 it will evaulate as
```
if (0 && retryCount++ >= 0)
```

So the first part (`0 && ..`) will always evaluate to false as `if (0)` is false

Reply:
So something like this would fix:
```suggestion
    if (!retryLimit || (retryLimit && retryCount++ >= retryLimit)) {
```

Reply:
I see. Forgot 0 was falsy. Also covers NaN which is falsy. Updated.